### ISSUE TITLE
plpgsql: add support for INTO statements with STRICT option

### ISSUE THREAD

**Issue by DrewKimball** [cockroachdb/cockroach#107854]
PLpgSQL `INTO` statements assign the result of executing a SQL statement to a specified set of target variables. Normally, the values from the first returned row are used, or NULLs if no rows are returned. However, it is possible to use `INTO STRICT` syntax to enforce that the SQL statement returns exactly one row. We should support this.

Example:
```
CREATE TABLE t (x INT);

CREATE FUNCTION f() RETURNS INT AS $$
  DECLARE
    i INT;
  BEGIN
    SELECT x INTO STRICT i FROM t;
    RETURN i;
  END 
$$ LANGUAGE PLpgSQL;

postgres=# SELECT f();
ERROR:  query returned no rows
CONTEXT:  PL/pgSQL function f() line 5 at SQL statement

INSERT INTO t VALUES (1);

postgres=# SELECT f();
 f
---
 1
(1 row)

INSERT INTO t VALUES (1);

postgres=# SELECT f();
ERROR:  query returned more than one row
HINT:  Make sure the query returns a single row, or use LIMIT 1.
CONTEXT:  PL/pgSQL function f() line 5 at SQL statement
```

**Comment by yuzefovich** [comment]
This is fixed by #120486, right @DrewKimball?

**Comment by DrewKimball** [comment]
Yes, I didn't realize this was a dupe.
